shared/src/main/java/uk/gov/di/authentication/shared/services/mfa/MfaMethodsService.java

```diff
@@ -299,4 +307,83 @@ private boolean updateRequestChangesMethodType(
                 ? !MFAMethodType.AUTH_APP.name().equals(methodTypeFromExisting)
                 : !MFAMethodType.SMS.name().equals(methodTypeFromExisting);
     }
+
+    public Optional<MfaMigrationFailureReason> migrateMfaCredentialsForUser(String email) {
+        // Bail if user doesn't exist
+        Optional<UserProfile> maybeUserProfile = persistentService.getUserProfileFromEmail(email);
+        Optional<UserCredentials> maybeUserCredentials =
+                Optional.ofNullable(persistentService.getUserCredentialsFromEmail(email));
+        if (maybeUserProfile.isEmpty() || maybeUserCredentials.isEmpty())
```

Comment:
mind adding brackets to these as adding an additional line by mistake changes the behaviour?